pennylane/resource/resource.py

```diff
@@ -31,6 +31,8 @@ class Resources:
         num_gates (int): number of gates
         gate_types (dict): dictionary storing operation names (str) as keys
             and the number of times they are used in the circuit (int) as values
+        gate_sizes (dict): dictionary storing the number of :math:`n` qubit gates in the circuit
+            as a key-value pair where :math:`n` is the key and the number of occurances is the value
```

Comment:
@Jaybsoni does this need to be something passed upon instantiation? Couldn't it be a property that is calculated based upon `gate_types`? I guess the issue is that `gate_types` holds strings as the keys (rather than the op itself), so we don't know easily know how many wires? :thinking:

Reply:
It will be passed on instantiation. Specs already computed/built this quantity from the tape, I just moved it into the Resources object. 

You are exactly right, if it's a custom operation (or an operation that takes `AnyWires`), we can't tell just from the string how many to count it as. Either way, it makes sense to have this as a property to be passed on instantiation to match the rest of the Resources fields.